code/game/objects/structures/flora.dm

```diff
@@ -263,6 +268,12 @@
 	var/matrix/M = matrix(transform)
 	transform = M.Turn(-previous_rotation)
 
+/obj/structure/flora/deconstruct()
+	if(!(flags_1 & NODECONSTRUCT_1))
+		harvest()
+		destroyed = TRUE
```

Comment:
If destroyed is only set to true after harvest is called, and the only place destroyed is checked is inside harvest, then wouldn't that mean destroying the flora would always drop the full amount of the results?